doc/source/ray-core/examples/batch_training.ipynb

```diff
@@ -529,66 +573,111 @@
     "    except Exception:\n",
     "        return []\n",
     "\n",
-    "    loc_list = locdf[\"pickup_location_id\"].unique()\n",
+    "    pickup_location_ids = locdf[\"pickup_location_id\"].unique()\n",
     "\n",
-    "    group_refs = []\n",
-    "    for i in loc_list:\n",
+    "    data_batch_refs = []\n",
+    "    for pickup_location_id in pickup_location_ids:\n",
     "        # Put each data batch as a separate dataframe into Ray object store.\n",
-    "        group_refs.append(\n",
-    "            (i, ray.put(locdf.filter(pc.field(\"pickup_location_id\") == i).to_pandas()))\n",
+    "        data_batch_refs.append(\n",
+    "            (\n",
+    "                pickup_location_id,\n",
+    "                ray.put(\n",
+    "                    locdf.filter(\n",
+    "                        pc.field(\"pickup_location_id\") == pickup_location_id\n",
+    "                    ).to_pandas()\n",
+    "                ),\n",
+    "            )\n",
     "        )\n",
     "\n",
-    "    return group_refs\n",
+    "    return data_batch_refs\n",
     "\n",
     "\n",
-    "def task_generator_with_object_store(\n",
+    "def run_batch_training_with_object_store(\n",
     "    files: List[str], models: List[BaseEstimator]\n",
-    ") -> ray.ObjectRef:\n",
-    "    # Use a placement group with a SPREAD strategy to load each\n",
+    "):\n",
+    "    print(\"Starting run...\")\n",
+    "    start = time.time()\n",
+    "\n",
+    "    # Store task references\n",
+    "    task_refs = []\n",
+    "\n",
+    "    # Use a SPREAD scheduling strategy to load each\n",
     "    # file on a separate node as an OOM safeguard.\n",
     "    # This is not foolproof though! We can also specify a resource\n",
     "    # requirement for memory, if we know what is the maximum\n",
     "    # memory requirement for a single file.\n",
-    "    pg = placement_group([{\"CPU\": 1}] * len(files), strategy=\"SPREAD\")\n",
-    "    ray.get(pg.ready())\n",
+    "    read_into_object_store_spread = read_into_object_store.options(\n",
+    "        scheduling_strategy=\"SPREAD\"\n",
+    "    )\n",
+    "    read_tasks = [read_into_object_store_spread.remote(file) for file in files]\n",
+    "\n",
+    "    # Dictionary of references to data batches with file names as keys\n",
+    "    data_batch_refs_by_file = {}\n",
+    "    for file_id, data_batch_refs in enumerate(ray.get(read_tasks)):\n",
```

Comment:
Not sure if we want to make it more complicated, but there is another optimization we can do here: we don't need to wait until all read_tasks to finish before submitting `train_and_evaluate`, as long as some read_task is finished, we can submit `train_and_evaluate`: https://docs.ray.io/en/master/ray-core/patterns/ray-get-submission-order.html

Reply:
I think it would make it too complex, yeah